### ISSUE TITLE
Integrate Weights & Biases for training run tracking

### ISSUE THREAD

**Issue by cemrtkn** [LBrinkmann/algorithmic-institutions#79]
## Goal

Log training metrics (train loss, per-fold test log_loss, feature shuffle/ablate deltas) to Weights & Biases so grid searches and hyperparameter sweeps get a sweep view (parallel coords, param importance, run comparison) instead of having to re-run the offline plotting scripts each time.

Reference pattern: [collectively-grounded-llms](https://github.com/center-for-humans-and-machines/collectively-grounded-llms) — uses env-var driven config (`WANDB_API_KEY`, `WANDB_ENTITY`, `WANDB_PROJECT`, `WANDB_RUN_GROUP`, `WANDB_NAME`). Their HF Trainer auto-initializes wandb; our `src/aimanager/artificial_humans/train.py` rolls its own loop so it will need explicit `wandb.init` / `wandb.log` / `wandb.finish`.

## Constraints

- **Not using djx.** SLURM scripts are generated by `src/aimanager/artificial_humans/run.py` (and manager/sim equivalents) from templates under `scripts/`. The wandb env-var plumbing has to go through those templates and the generator.
- Raven compute nodes have internet — online mode works, no offline/sync dance needed.
- Existing configs must keep working if wandb env vars are not set (integration is opt-in).

### 1. Secrets and `.env`

- [ ] Create `.env.example` with `WANDB_API_KEY=<your wandb api key>`.
- [ ] Ensure `.env` is in `.gitignore` (check; add if missing).
- [ ] Document in README how to populate `.env` on Raven (copy API key from existing setup in another project).

### 2. SLURM templates

Add wandb env-var exports to each SLURM template (source `.env` and export the five `WANDB_*` variables before running the command):

- [ ] `scripts/artificial_humans/run_training.sh`
- [ ] `scripts/manager/run_training.sh` (or equivalent; confirm path)
- [ ] `scripts/run_simulation.sh` — only if we also want to log simulation runs

Variables to export:

```bash
source .env
export WANDB_API_KEY
export WANDB_ENTITY=chm-ml
export WANDB_PROJECT=algorithmic-institutions
export WANDB_RUN_GROUP={experiment_name}  # filled in by generator
export WANDB_NAME={job_id}
```

### 3. Job generator

- [ ] Update `src/aimanager/artificial_humans/run.py` (and manager/sim equivalents) to pass `experiment_name` / `job_id` placeholders into the template. `experiment_name` should be derivable from the config path the same way the other repo does it (e.g. the top-level directory name under `configs/`).

### 4. Training code

- [ ] `pyproject.toml` — add `wandb` dependency and run `uv sync`.
- [ ] `src/aimanager/artificial_humans/train.py` — at `main(config)` start, if `os.environ.get(\"WANDB_API_KEY\")`, call `wandb.init(config=config)`.
- [ ] Mirror every `rec.rec(...)` / `rec.rec_many(...)` with an equivalent `wandb.log({...})`. Key naming: `train/loss`, `fold_{i}/test/log_loss`, plus aggregated `test/log_loss_mean` at each eval epoch.
- [ ] Call `wandb.finish()` at end of training.
- [ ] **CV folds: log all folds to one wandb run** (not one per fold) — since djx is not used and all folds run in a single SLURM job, one wandb run per job = one point per grid-search config in the sweeps view.

### 5. Config hook (optional but nice)

- [ ] Add optional `wandb:` block in training YAMLs for per-experiment overrides (tags, project override). Default to env-driven behavior if absent.

### 6. Manual test

- [ ] Run a small job (e.g. 500 epochs of the 35ep config) with `WANDB_API_KEY` set and verify the run shows up on wandb.
- [ ] Run the grid search and confirm the sweeps view renders 9 runs with parallel-coords and hyperparameter importance.

## Out of scope for this issue

- Switching existing plotting scripts (`plot_cv_summary.py`) to consume wandb API instead of parquet files. Parquet stays canonical; wandb is the online view.
- Logging to wandb from the simulation pipeline (separate step after training integration is stable).

## Success criteria

Submitting a grid search to Raven produces N wandb runs (one per grid-search job) under the `algorithmic-institutions` project, each showing train/test loss curves and feature shuffle/ablate deltas, with a sweeps view available for comparing configs.
